Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1031,6 +1054,30 @@ public extension Purchases {
 
 // swiftlint:enable missing_docs
 
+// MARK: - Paywalls
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+public extension Purchases {
+
+    /// Used by `RevenueCatUI` to keep track of ``PaywallEvent``s.
+    func track(paywallEvent: PaywallEvent) async {
+        switch paywallEvent {
+        case let .view(data):
+            self.purchasesOrchestrator.cachePresentedPaywall(data)
+
+        case .close:
+            self.purchasesOrchestrator.clearPresentedPaywall()
+
+        case .cancel:
+            // No special handling, simply track the event below.
+            break
+        }
+
+        await self.paywallEventsManager?.track(paywallEvent: paywallEvent)
```

Comment:
should we be moving this to the orchestrator layer? Ideally we do as little as possible at the Facade layer

Reply:
This has nothing to do with _purchases_ though. `PaywallEventsManager` was my way of extracting this out, though I agree ideally methods in `Purchases` are no more than 1 line.